lsp-mode.el

```diff
@@ -4062,8 +4070,7 @@ The method uses `replace-buffer-contents'."
                     (length (- end beg)))
                 (run-hook-with-args 'before-change-functions
                                     beg end)
-                (when (fboundp 'replace-buffer-contents)
-                  (with-no-warnings (replace-buffer-contents temp)))
+                (replace-buffer-contents temp)
```

Comment:
Just asking but what's the advantage of this `replace-buffer-contents` over `delete` and `insert`?
I've done some simple benchmark and it seems the the later one is much faster.
``` elisp
(with-current-buffer (get-buffer-create "xyz")
  (insert "1234\n123\n123333"))
(with-current-buffer (get-buffer-create "aa")
  (insert "1234\n123\n123333\n1234\n123\n123333"))

(benchmark-run 100000 '(with-current-buffer (get-buffer-create "xyz")
                       (replace-buffer-contents (get-buffer-create "aa"))))
(0.017054 0 0.0)

(benchmark-run 100000 '(with-current-buffer (get-buffer-create "xyz")
                         (delete-region (point-min) (point-max))
                         (insert (with-current-buffer (get-buffer-create "aa") (buffer-string)))))
(0.0014909999999999993 0 0.0)
```

Reply:
replace-buffer-contents does not break lenses and overlays.

Reply:
And also it is faster in real-world scenarios due to the fact that it takes advantage from the internal buffer representation. E. g. apply 100+ edits in 1mb file.